idaes/models_extra/temperature_swing_adsorption/fixed_bed_tsa0d.py

```diff
@@ -0,0 +1,3677 @@
+            initialize=1e3, units=units.s, doc="Time of heating step"
+        )
+        self.heating.time_domain = ContinuousSet(
+            bounds=(0, 1), doc="Normalized time domain in heating step"
+        )
+
+        # variables
+        def mole_frac_init(b, t, i):
+            return self.mole_frac_in[i]
+
+        self.heating.mole_frac = Var(
+            self.heating.time_domain,
+            self.isotherm_components,
+            bounds=(0, 1),
+            initialize=mole_frac_init,
+            units=units.dimensionless,
+            doc="Mole fraction in heating step",
+        )
+        self.heating.temperature = Var(
+            self.heating.time_domain,
+            initialize=self.temperature_adsorption,
+            units=units.K,
+            doc="Temperature in heating step",
+        )
+        self.heating.velocity_out = Var(
+            self.heating.time_domain,
+            initialize=self.velocity_in,
+            units=units.m / units.s,
+            doc="Outlet velocity in heating step",
+        )
+        self.heating.loading = Var(
+            self.heating.time_domain,
+            self.isotherm_components,
+            units=units.mol / units.kg,
+            doc="Equilibrium loading in heating step",
+        )
+
+        # time derivatives
+        self.heating.mole_frac_dt = DerivativeVar(
+            self.heating.mole_frac,
+            wrt=self.heating.time_domain,
+            units=units.dimensionless,
+            doc="Time derivative of mole fraction in heating step",
+        )
+        self.heating.temperature_dt = DerivativeVar(
+            self.heating.temperature,
+            wrt=self.heating.time_domain,
+            units=units.K,
+            doc="Time derivative of temperature in heating step",
+        )
+        self.heating.loading_dt = DerivativeVar(
+            self.heating.loading,
+            wrt=self.heating.time_domain,
+            units=units.mol / units.kg,
+            doc="Time derivative of equilibrium loading in heating step",
+        )
+
+        # partial pressure
+        @self.heating.Expression(
+            self.heating.time_domain,
+            self.isotherm_components,
+            doc="Partial pressure in heating step [Pa]",
+        )
+        def partial_pressure(b, t, i):
+            return self._partial_pressure(b.pressure, b.mole_frac[t, i])
+
+        # sum of mole fractions
+        @self.heating.Constraint(
+            self.heating.time_domain, doc="Sum of mole fraction constraint"
+        )
+        def sum_mole_frac(b, t):
+            return sum(b.mole_frac[t, j] for j in self.isotherm_components) == 1.0
+
+        # component mass balance ODE
+        @self.heating.Constraint(
+            self.heating.time_domain,
+            self.isotherm_components,
+            doc="Component mass balance - only for CO2",
+        )
+        def component_mass_balance_ode(b, t, i):
+            if t == b.time_domain.first() or i == "N2":
```

Comment:
A minor word of caution, but this does presume the presence of N2 in the system. For a DAC system this is to be expected, and is probably a good approach to improve robustness (my intuition tells me an overall mass balance will be easier to solve that having to solve it implicitly via the full set of component balances).

However, it does result in a loss of generality (which probably doesn't matter for now), in that the model cannot be used for systems without N2.